tests/integration/pilot/istioctl_test.go

```diff
@@ -391,8 +393,10 @@ func TestAuthZCheck(t *testing.T) {
 		RequiresSingleCluster().
 		Run(func(ctx framework.TestContext) {
 			authPol := file.AsStringOrFail(t, "testdata/authz-a.yaml")
-			ctx.Config().ApplyYAMLOrFail(t, apps.namespace.Name(), authPol)
-			defer ctx.Config().DeleteYAMLOrFail(t, apps.namespace.Name(), authPol)
+			ctx.Config().ApplyYAMLOrFail(ctx, apps.namespace.Name(), authPol)
+			ctx.WhenDone(func() error {
```

Comment:
not really that related to your PR, but we should have a way to skip cleanup of apps and skip cleanup of config separably. Otherwise its not really feasible to use `nocleanup` flag

Reply:
which is why most of these use defer

Reply:
Ah good point. I rolled https://github.com/istio/istio/pull/26530 which should address that.